src/Compilers/VisualBasic/Portable/Compilation/VisualBasicDeterministicKeyBuilder.vb

```diff
@@ -0,0 +1,75 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Namespace Microsoft.CodeAnalysis.VisualBasic
+
+    Friend NotInheritable Class VisualBasicDeterministicKeyBuilder
+        Inherits DeterministicKeyBuilder
+
+        Public Sub New()
+
+        End Sub
```

Comment:
this type appears to be a singleton.  Consider making constructors prviate and expose a single static instance.

Reply:
same applies to C#.